### PR TITLE
feat: add support for angular 17

### LINKED ISSUE #28153: Add Angular 17 Support for CT
### What would you like?

Angular 17 is set to launch on November 6. Ideally we will have support in CT on day 1.

### Why is this needed?

Supporting the newest versions of our frameworks is crucial for overall adoption and trust in the long term state of CT

### Other

_No response_

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes https://github.com/cypress-io/cypress/issues/28153 and https://github.com/cypress-io/cypress/issues/27475

### Additional details
This PR adds support for Angular 17 CT set to release on November 7. AFAIK their are no breaking changes and just requires a bump in the scaffold deps + system tests example for version 17. This PR currently uses the Angular 17 **Release Candidate** to validate it works. We will need to change this before merging

### Steps to test
Run `yarn dev` and open an angular 17 project using using the Release candidate. 

### How has the user experience changed?
No changes expected

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jordanpowell88** [comment]
Duplicate of https://github.com/cypress-io/cypress-internal/issues/704

File: packages/scaffold-config/src/dependencies.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export const WIZARD_DEPENDENCY_ANGULAR_CLI = {
   package: '@angular/cli',
   installer: '@angular/cli',
   description: 'CLI tool that you use to initialize, develop, scaffold, and maintain Angular applications.',
-  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0',
+  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0 || ^=17.0.0',
```
Comment (MikeMcC399) [inline packages/scaffold-config/src/dependencies.ts]:
```suggestion
  minVersion: '^13.0.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0',
```

File: packages/scaffold-config/src/dependencies.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ export const WIZARD_DEPENDENCY_ANGULAR_DEVKIT_BUILD_ANGULAR = {
   package: '@angular-devkit/build-angular',
   installer: '@angular-devkit/build-angular',
   description: 'Angular Webpack build facade',
-  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0',
+  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0 || ^=17.0.0',
```
Comment (MikeMcC399) [inline packages/scaffold-config/src/dependencies.ts]:
```suggestion
  minVersion: '^13.0.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0',
```

File: packages/scaffold-config/src/dependencies.ts [hunk 3]
```diff
@@ -121,7 +121,7 @@ export const WIZARD_DEPENDENCY_ANGULAR_CORE = {
   package: '@angular/core',
   installer: '@angular/core',
   description: 'The core of the Angular framework',
-  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0',
+  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0 || ^=17.0.0',
```
Comment (MikeMcC399) [inline packages/scaffold-config/src/dependencies.ts]:
```suggestion
  minVersion: '^13.0.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0',
```

File: packages/scaffold-config/src/dependencies.ts [hunk 4]
```diff
@@ -130,7 +130,7 @@ export const WIZARD_DEPENDENCY_ANGULAR_COMMON = {
   package: '@angular/common',
   installer: '@angular/common',
   description: 'Commonly needed Angular directives and services',
-  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0',
+  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0 || ^=17.0.0',
```
Comment (MikeMcC399) [inline packages/scaffold-config/src/dependencies.ts]:
```suggestion
  minVersion: '^13.0.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0',
```

File: packages/scaffold-config/src/dependencies.ts [hunk 5]
```diff
@@ -139,7 +139,7 @@ export const WIZARD_DEPENDENCY_ANGULAR_PLATFORM_BROWSER_DYNAMIC = {
   package: '@angular/platform-browser-dynamic',
   installer: '@angular/platform-browser-dynamic',
   description: 'Library for using Angular in a web browser with JIT compilation',
-  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0',
+  minVersion: '^=13.0.0 || ^=14.0.0 || ^=15.0.0 || ^=16.0.0 || ^=17.0.0',
```
Comment (MikeMcC399) [inline packages/scaffold-config/src/dependencies.ts]:
```suggestion
  minVersion: '^13.0.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0',
```

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
The syntax `^=` happens to work although it is incorrect and has been perpetuated through copy and paste.

If code lines are being changed, then this is a good opportunity to correct it.

- See #27475

The proper syntax is described in [Caret ranges](https://github.com/npm/node-semver#caret-ranges-123-025-004).

**Comment by MikeMcC399** [comment]
@jordanpowell88 

You need to write 

- closes #27475

separately in the original PR text, otherwise GitHub doesn't pick it up and won't link it automatically, to close it when the PR is closed.

image

from https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Review by MikeMcC399 — APPROVED** [review]
Thanks for changing the syntax!

**Comment by MikeMcC399** [comment]
@jordanpowell88 

According to https://angular.io/guide/versions only Angular `14`, `15`, and `16` are actively supported by Angular, so `13` should probably be removed.

Also Angular `14` requires Node.js `14` or `16` and Cypress no longer supports these versions of Node.js (see https://docs.cypress.io/guides/getting-started/installing-cypress#Nodejs), so Angular `14` should probably also be removed from Cypress support.

Since removal would be a breaking change this might need to be reserved for a later Cypress major version release. (It would however be possible to remove the old system tests and leave the version check without causing a breaking change.)

File: system-tests/projects/angular-16.1/package.json
Comment (jordanpowell88) [inline system-tests/projects/angular-16.1/package.json:1]: We don't need system tests for both `angular-16` and `angular-16.1` anymore.

File: system-tests/projects/angular-17/cypress.config.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            '@angular/common/http': require.resolve('@angular/common/http'),
+            '@angular/common/testing': require.resolve('@angular/common/testing'),
+            '@angular/common': require.resolve('@angular/common'),
+            '@angular/core/testing': require.resolve('@angular/core/testing'),
+            '@angular/core/primitives/signals': require.resolve('@angular/core/primitives/signals'),
+            '@angular/core': require.resolve('@angular/core'),
+            '@angular/platform-browser/testing': require.resolve('@angular/platform-browser/testing'),
+            '@angular/platform-browser': require.resolve('@angular/platform-browser'),
+            '@angular/platform-browser-dynamic/testing': require.resolve('@angular/platform-browser-dynamic/testing'),
+            '@angular/platform-browser-dynamic': require.resolve('@angular/platform-browser-dynamic'),
```
Comment (jordanpowell88) [inline system-tests/projects/angular-17/cypress.config.ts:21]: Changes in angular 17  to `zone.js/testing` cause the shared `cypress.config.ts` from `project-fixtures/angular/cypress.config.ts` to be invalid so needed create a unique one here

**Review by astone123 — APPROVED** [review]
nice! 🎉

File: cli/CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,10 @@
 
 _Released 11/7/2023 (PENDING)_
 
+**Features:**
+
+ - Adds Component Testing support for Angular version 17. Addresses [#28153](https://github.com/cypress-io/cypress/issues/28153).
+
```
Comment (mike-plummer) [inline cli/CHANGELOG.md:9]: Since this is a feature  we should bump the next version number to 13.5.0, right?

**Review by mike-plummer — APPROVED** [review]
One comment on versioning, otherwise looks good

File: cli/CHANGELOG.md (hunk 7 above)
Comment (jordanpowell88) [reply cli/CHANGELOG.md:9]: Yup

Comment (emilyrohrbough) [reply cli/CHANGELOG.md:9]: This check should have ensured this was bumped before merging...

**Pull request merged**